File: frontend/mtguardian/src/lib/types/Prescription.ts
interface IObjectKeys {
  [key: string]: string | number | null;
}

export interface Prescription extends IObjectKeys {
  id: string;
  medication: string;
  dosage: string;
  notes: string;
  started: string | null;
  ended: string | null;
}

export const generatePrescriptionTemplate = () => {
  return {
    id: "",
    medication: "",
    dosage: "",
    notes: "",
    started: "",
    ended: "",
  };
};

File: backend/gateway/internal/customtype/prescription.go
package customtype

import (
	"time"

	"github.com/google/uuid"
)

type Prescription struct {
	ID         uuid.UUID  `json:"id" gorm:"type:uuid;primaryKey"`
	Medication *string    `json:"medication"`
	Dosage     *string    `json:"dosage"`
	Notes      *string    `json:"notes"`
	Started    *time.Time `json:"started" gorm:"type:timestamp;"`
	Ended      *time.Time `json:"ended" gorm:"type:timestamp;"`
}
